primo/utils/override_utils.py

```diff
@@ -12,224 +12,143 @@
 
     def assess_owner_well_count(self) -> Dict:
+        # pylint: disable=protected-access
         """
         Assess whether the owner well count constraint is violated or not.
         Returns list of owners and wells selected for each for whom the owner
         well count constraint is violated
         """
         violated_operators = {}
-        for operator, groups in self._plugged_list.groupby("Operator Name"):
+        for operator, groups in self.wd.data.groupby(self.wd._col_names.operator_name):
             n_wells = len(groups)
-            if n_wells > self.opt_inputs.max_wells_per_owner:
-                violated_operators[operator] = [
-                    n_wells,
-                    groups["API Well Number"].to_list(),
-                ]
+            if n_wells > self.opt_inputs.config.max_wells_per_owner:
+                violated_operators.setdefault("Owner", []).append(operator)
+                violated_operators.setdefault("Number of wells", []).append(n_wells)
+                violated_operators.setdefault("Wells", []).append(
+                    groups[self.wd._col_names.well_id].to_list()
+                )
 
         return violated_operators
 
     def assess_distances(self) -> Dict:
+        # pylint: disable=protected-access
         """
         Assess whether the maximum distance between two wells constraint is violated or not
         """
-        distance_threshold = self.opt_inputs.distance_threshold
-        distance_violations = {}
-        # TODO: Consider calculating these once and for all since back filling
-        # might end up causing multiple iterations here
-        for cluster_id, groups in self._plugged_list.groupby("Project"):
-            well_id = list(groups["API Well Number"])
-            num_well = len(well_id)
-            groups["coords"] = list(zip(groups.Latitude, groups.Longitude))
-            distance_metric_distance = haversine_vector(
-                groups["coords"].to_list(),
-                groups["coords"].to_list(),
-                unit=Unit.MILES,
-                comb=True,
-            )
-            well_distance = {}
-            for well_1 in range(num_well - 1):
-                for well_2 in range(well_1 + 1, num_well):
-                    well_distance = distance_metric_distance[well_1][well_2]
-                    if well_distance > distance_threshold:
-                        distance_violations[
-                            (cluster_id, well_id[well_1], well_id[well_2])
-                        ] = [
-                            cluster_id,
-                            well_id[well_1],
-                            well_id[well_2],
-                            well_distance,
-                        ]
-
-        return distance_violations
+        distance_threshold = self.opt_inputs.config.threshold_distance
+        distance_violation = {}
+        metric_array = distance_matrix(self.wd, {"distance": 1})
+        df_to_array = {
```

Comment:
This is going to have to change in the future when we go larger scale. Also, is there a reason the 1 is hardcoded? Do we have a parameter for distance that we pass in?

Reply:
The `distance_matrix` is utilized for clustering and requires specifying weights for distance, age, and depth. In this instance, all weight is assigned to distance, resulting in a value of 1 in the code.